lib/src/androidTest/java/com/couchbase/lite/ResultTest.java

```diff
@@ -1082,4 +1082,23 @@ public void check(int n, Result r) throws Exception {
             assertEquals(1, rows);
         }
     }
+
+    // Contributed by Bryan Welter:
+    // https://github.com/couchbase/couchbase-lite-android-ce/issues/27
+    @Test
+    public void testEmptyDictionary() throws Exception {
+        String doc1 = "doc1";
+        String key1 = "emptyDict";
+
+        MutableDocument mDoc = new MutableDocument(doc1);
+        mDoc.setDictionary(key1, new MutableDictionary());
+        save(mDoc);
+
+        final Query query = QueryBuilder.select(SelectResult.property(key1))
+            .from(DataSource.database(db))
+            .where(Meta.id.equalTo(Expression.string(doc1)));
+
+        ResultSet results = query.execute();
+        for (Result result : results.allResults()) { result.toMap(); }
```

Comment:
Should be assert if result.toMap() is empty?

Reply:
Wooo... good idea.  It is the `allResults()` that used to fail.  Still, that's a good test.  Will add.